framework/src/modules/interoperability/mainchain/module.ts

```diff
@@ -419,4 +438,108 @@ export class MainchainInteroperabilityModule extends BaseInteroperabilityModule
 			throw new Error('Invalid validatorsHash from chainData.lastCertificate.');
 		}
 	}
+
+	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#mainchain
+	private _verifyTerminatedStateAccounts(
+		chainInfos: ChainInfo[],
+		terminatedStateAccounts: TerminatedStateAccountWithChainID[],
+	) {
+		for (const chainInfo of chainInfos) {
+			const { chainID } = chainInfo;
+
+			const filteredTerminatedStateAccounts = terminatedStateAccounts.filter(a =>
+				a.chainID.equals(chainID),
+			);
+			if (terminatedStateAccounts.length > 0 && filteredTerminatedStateAccounts.length === 0) {
+				throw new Error('there can not be a terminated account if there is no chain account');
```

Comment:
maybe we can simplify this,
- Iterate over chainInfos
- Check if a chainInfo.chainData.status === CHAIN_STATUS_TERMINATED then `const terminatedAccount = terminatedStateAccounts.find(tAccount => tAccount.chainID.equals(chainInfo.chainData.chainID))`
- If `terminatedAccount` is undefined then throw error that `For each chainInfo with status terminated there should be a corresponding entry in terminatedStateAccounts`